x-pack/solutions/observability/plugins/observability/public/components/custom_threshold/custom_threshold_rule_expression.tsx

```diff
@@ -99,6 +104,32 @@ export default function Expressions(props: Props) {
     [dataView]
   );
 
+  const fieldsQuery = useFetchAlertsFieldsQuery(
+    {
+      http,
+      ruleTypeIds: [OBSERVABILITY_THRESHOLD_RULE_TYPE_ID],
+    },
+    {
+      enabled: true,
+    }
+  );
+
+  useEffect(() => {
+    const groupingFields = Object.keys(
+      fieldsQuery.data?.browserFields?.kibana?.fields || {}
+    ).filter((f) => {
+      return f.startsWith('kibana.alert.grouping');
+    });
+
+    if ((ruleParams.groupBy || []).length + groupingFields.length > MAX_SELECTABLE_RUNTIME_FIELDS) {
+      setGroupByError(
+        `You can't select more than ${MAX_SELECTABLE_RUNTIME_FIELDS - groupingFields.length} groups`
+      );
+    } else {
+      setGroupByError(undefined);
+    }
+  }, [fieldsQuery, ruleParams.groupBy]);
+
```

Comment:
Add this FE side validation that uses the existing fields query.
We may provide an api that returns only the runtime fields, so it can be used for validation here.

Note: There is no way to pass async validation params to the validation function on the rule type, therefore those validations does not disable the save button on the rule create/edit form. Maybe we should create an issue for this improvement.